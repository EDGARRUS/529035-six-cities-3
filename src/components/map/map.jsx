import React, {PureComponent} from "react";
import leaflet from "leaflet";
import PropTypes from "prop-types";

export class Map extends PureComponent {
  constructor(props) {
    super(props);
    this.map = React.createRef();
  }
  render() {
    return <section id="map" ref={this.map} className="cities__map map"></section>;
  }
  componentDidMount() {
    if (this.map.current) {
      const offersData = this.props.offersData;
      const city = [52.38333, 4.9];
      const icon = leaflet.icon({
        iconUrl: `img/pin.svg`,
        iconSize: [30, 30]
      });
      const zoom = 12;
      const map = leaflet.map(`map`, {
        center: city,
        zoom,
        zoomControl: false,
        marker: true
      });
      map.setView(city, zoom);
      offersData.map((offer) => leaflet.marker(offer.coordinates, {icon}).addTo(map));
      leaflet.tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`}).addTo(map);
    }
  }
}

Map.propTypes = {
  offersData: PropTypes.arrayOf(PropTypes.exact({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    coordinates: PropTypes.arrayOf(PropTypes.number).isRequired,
    gallery: PropTypes.arrayOf(PropTypes.string).isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    rate: PropTypes.number,
    isPremium: PropTypes.bool.isRequired,
    description: PropTypes.string,
    bedrooms: PropTypes.number.isRequired,
    maxGuests: PropTypes.number.isRequired,
    devices: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.exact({
      avatar: PropTypes.string,
      name: PropTypes.string.isRequired,
      isSuper: PropTypes.bool.isRequired,
    })
  })).isRequired,
};
