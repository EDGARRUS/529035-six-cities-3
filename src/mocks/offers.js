export const offersData = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    image: `img/apartment-01.jpg`,
    isPremium: true,
    price: 120,
    type: `apartment`,
    rate: 4,
  },

  {
    title: `Wood and stone place`,
    image: `img/apartment-02.jpg`,
    isPremium: false,
    price: 80,
    type: `room`,
    rate: 5,
  },

  {
    title: `Canal View Prinsengracht`,
    image: `img/apartment-03.jpg`,
    isPremium: false,
    price: 132,
    type: `house`,
    rate: 3,
  },

  {
    title: `Nice, cozy, warm big bed apartment`,
    image: `img/apartment-01.jpg`,
    isPremium: false,
    price: 56,
    type: `hotel`,
    rate: 2.5,
  },

];
