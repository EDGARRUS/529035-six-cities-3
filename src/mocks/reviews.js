export const reviewsData = [
  {
    id: 1,
    avatar: `img/avatar-max.jpg`,
    name: `Masha`,
    userRate: 4,
    dateReview: `December 24, 2018`,
    description: `Очень хороший отель мне понравилось!`,
    offerId: 1,
  },
  {
    id: 2,
    avatar: `img/avatar-max.jpg`,
    name: `Sergey`,
    userRate: 3,
    dateReview: `December 24, 2018`,
    description: `Никому не рекомендую. Очень дорого и грязно`,
    offerId: 1,
  },
  {
    id: 3,
    avatar: `img/avatar-max.jpg`,
    name: `Dima`,
    userRate: 2,
    dateReview: `December 24, 2018`,
    description: `Дважды подумайте, если будете бронировать это место!`,
    offerId: 2,
  },
  {
    id: 4,
    avatar: `img/avatar-max.jpg`,
    name: `Eduard`,
    userRate: 1,
    dateReview: `December 24, 2018`,
    description: `Ужасное и страшное место, я испугалась тараканов`,
    offerId: 3,
  },
  {
    id: 5,
    avatar: `img/avatar-max.jpg`,
    name: `Mike`,
    userRate: 5,
    dateReview: `December 24, 2018`,
    description: `Шикарный отел с бассейном и бутылкой рома`,
    offerId: 3,
  },
];
