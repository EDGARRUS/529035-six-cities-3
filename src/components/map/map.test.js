import React from "react";
import renderer from "react-test-renderer";
import {Map} from "./map.jsx";
import {offersData} from "../../mocks/offers";

it(`<Map /> Testing`, () => {
  const tree = renderer.create(<Map offersData={offersData}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
