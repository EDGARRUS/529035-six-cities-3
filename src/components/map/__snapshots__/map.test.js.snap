// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> Testing 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
